core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -229,41 +227,56 @@ class DAGScheduler(
 
   /**
    * Create a Stage -- either directly for use as a result stage, or as part of the (re)-creation
-   * of a shuffle map stage in newOrUsedStage.  The stage will be associated with the provided
-   * jobId. Production of shuffle map stages should always use newOrUsedStage, not newStage
-   * directly.
+   * of a shuffle map stage in newOrUsedShuffleStage.  The stage will be associated with the provided
+   * jobId. Production of shuffle map stages should always use newOrUsedShuffleStage, 
+   * not getAndRegNewShuffleMap directly.
    */
-  private def newStage(
-      rdd: RDD[_],
-      numTasks: Int,
-      shuffleDep: Option[ShuffleDependency[_, _, _]],
-      jobId: Int,
-      callSite: CallSite)
-    : Stage =
-  {
+  private def getAndRegNewShuffleMap(rdd: RDD[_],
+                                     numTasks: Int,
+                                     shuffleDep: ShuffleDependency[_, _, _],
+                                     jobId: Int,
+                                     callSite: CallSite): ShuffleMapStage = {
     val parentStages = getParentStages(rdd, jobId)
-    val id = nextStageId.getAndIncrement()
-    val stage = new Stage(id, rdd, numTasks, shuffleDep, parentStages, jobId, callSite)
+    val id = nextStageId.getAndIncrement
+    val stage: ShuffleMapStage = new ShuffleMapStage(id, rdd, numTasks, parentStages, 
+      jobId, callSite, shuffleDep)
+      
     stageIdToStage(id) = stage
     updateJobIdStageIdMaps(jobId, stage)
     stage
   }
 
   /**
+   * Create a Stage -- either directly for use as a result stage, or as part of the (re)-creation
+   * of a shuffle map stage in newOrUsedShuffleStage.  The stage will be associated with the provided
+   * jobId. Production of shuffle map stages should always use newOrUsedShuffleStage, 
+   * not getAndRegNewResults directly.
+   */
+  private def getAndRegNewResults(rdd: RDD[_],
```

Comment:
I'm not a fan of the `getAndReg` addition since `get` falsely implies some kind of look up, and `reg` is more-or-less an implementation detail that across future refactorings won't necessarily be a fixed part of the creation of new stages.  The new, longer naming just doesn't add anything positive, imo.